src/python/pants/backend/python/tasks/select_interpreter.py

```diff
@@ -95,6 +98,17 @@ def _create_interpreter_path_file(self, interpreter_path_file, targets):
   def _interpreter_path_file(self, target_set_id):
     return os.path.join(self.workdir, target_set_id, 'interpreter.info')
 
+  def _detect_and_purge_invalid_interpreter(self, interpreter_path_file):
+    with open(interpreter_path_file, 'r') as infile:
```

Comment:
Instead of re-creating the file reading logic, how about something like:
```python
interpreter = self._get_interpreter(interpreter_path_file)
if not os.path.exists(os.readlink(interpreter.binary)):
  ...
```